src/node/server/serverPluginModuleRewrite.ts

```diff
@@ -254,11 +252,8 @@ export const resolveImport = (
     //    ./foo -> /some/path/foo
     let { pathname, query } = resolveRelativeRequest(importer, id)
 
-    // 2. resolve extensions.
-    const ext = resolver.resolveExt(pathname)
```

Comment:
This is the cause of bug. resolveExt is misused.
for pathname `/test-package/lib/dir`, resolveExt returned `.js` instead of `/index.js` here.
So it is rewrited into `/test-package/lib/dir.js` instead of `/test-package/lib/dir/index.js`